### PR TITLE
添加 line shape 的测试用例及一系列 BUGFIX

### PR DESCRIPTION
**Pull request by simaQ**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [ ] All tests pass and/or benchmarks are included
- [ ] Commit message follows commit guidelines
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: src/geometry/animate/util.ts [hunk 1]
```diff
@@ -125,7 +125,7 @@ export function getCoordinateClipCfg(coordinate: Coordinate) {
 
 export function getAngle(shapeModel: ShapeDrawCFG, coordinate: Coordinate) {
   const points = shapeModel.points;
-  const box = getPointsBox(points);
+  const box = getPointsBox(points as Point[]);
```
Comment (hustcc) [inline src/geometry/animate/util.ts]: 看 ShapeDrawCFG 的类型定义，应该不需要 as 啊~~~

Comment (simaQ) [reply src/geometry/animate/util.ts]: oops，已解

File: src/geometry/shape/util/get-path-points.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * splitData([{x: 1, y: 2}, {x: 2, y: null}, {x: 3, y: 3}], 'y', true);
+ * @example
+ * // return [[{x: 1, y: 2}], [{x: 3, y: 3}]]
+ * splitData([{x: 1, y: 2}, {x: 2, y: null}, {x: 3, y: 3}], 'y', false);
+ *
+ * @param data 要进行处理的数据
+ * @param field 判断空值的字段名
+ * @param connectNulls 是否连接空值数据
+ */
+export function getPathPoints(points: Point[] | RangePoint[], connectNulls?: boolean) {
```
Comment (hustcc) [inline src/geometry/shape/util/get-path-points.ts]: RangePoint 包含有 Point，无需用 或

Comment (simaQ) [reply src/geometry/shape/util/get-path-points.ts]: done

**Pull request merged**